### PR TITLE
[FIX] used numerical indices for references

### LINKED ISSUE #2185: Inconsistency in stating references in dti.py
## Description
Inconsistency in stating references

In the docstring of restore_fit_tensor function, there was an inconsistency in the way References were attributed. Everywhere else numeric indices were used as placeholders for References, but here Chang2005 was used.

https://github.com/dipy/dipy/blob/5e4adff8b1d86210a683b84034c18e388914a09e/dipy/reconst/dti.py#L1740

https://github.com/dipy/dipy/blob/5e4adff8b1d86210a683b84034c18e388914a09e/dipy/reconst/dti.py#L1754

https://github.com/dipy/dipy/blob/5e4adff8b1d86210a683b84034c18e388914a09e/dipy/reconst/dti.py#L1773

### PR DESCRIPTION
**Pull request by areeshatariq**
fixes #2185

### CODE REVIEW

File: dipy/reconst/dti.py [hunk 1]
```diff
@@ -1770,7 +1770,7 @@ def restore_fit_tensor(design_matrix, data, sigma=None, jac=True,
 
     References
     ----------
-    Chang, L-C, Jones, DK and Pierpaoli, C (2005). RESTORE: robust estimation
+    [1] Chang, L-C, Jones, DK and Pierpaoli, C (2005). RESTORE: robust estimation
```
Comment (arokem) [inline dipy/reconst/dti.py:1773]: I believe this should be `.. [1]` for this to be properly linked and rendered by sphinx.

**Review by arokem — COMMENTED** [review]
Looks good! I added one comment that would fix a problem in the existing doc.

File: dipy/reconst/dti.py (hunk 1 above)
Comment (areeshatariq) [reply dipy/reconst/dti.py]: @arokem yes, you're right. Custom anchors should be created for linking the references. They were missing for some other references too.

**Comment by jhlegarreta** [comment]
Nice @areeshatariq ! :+1:

The following have to do more with the commit/PR style and some aspects that may help your PRs shine even more :sparkles: :
- When an in-line review is addressed, marking it the conversation as resolved :heavy_check_mark: would be desirable . This allows to keep the PR cleaner and helps other reviewers focus on unresolved issues :book:.
- The commit message subject has the prefix `[DOC]` whereas the PR has the prefix `[FIX]`. Either would be fine in this case, but when a PR has a single commit it is probably easier to use the same for the sake of consistency :heavy_check_mark: .
- Although commit messages are usually not consistent concerning some formal aspects, I find [these guidelines](https://chris.beams.io/posts/git-commit/) useful to write messages that can help better reviewers and the history of the project. This is an ever lasting debate, and mechanisms vary across projects, but you may find interesting discussions about these aspects in [this](https://itk.org/pipermail/insight-developers/2010-October/016008.html) and [this other thread](https://vtk.org/pipermail/vtk-developers/2016-May/033505.html). Reading them is worthwhile IMHO.

Good job.

**Comment by areeshatariq** [comment]
> Nice @areeshatariq ! +1
> 
> The following have to do more with the commit/PR style and some aspects that may help your PRs shine even more sparkles :
> 
>     * When an in-line review is addressed, marking it the conversation as resolved heavy_check_mark would be desirable . This allows to keep the PR cleaner and helps other reviewers focus on unresolved issues book.
> 
>     * The commit message subject has the prefix `[DOC]` whereas the PR has the prefix `[FIX]`. Either would be fine in this case, but when a PR has a single commit it is probably easier to use the same for the sake of consistency heavy_check_mark .
> 
>     * Although commit messages are usually not consistent concerning some formal aspects, I find [these guidelines](https://chris.beams.io/posts/git-commit/) useful to write messages that can help better reviewers and the history of the project. This is an ever lasting debate, and mechanisms vary across projects, but you may find interesting discussions about these aspects in [this](https://itk.org/pipermail/insight-developers/2010-October/016008.html) and [this other thread](https://vtk.org/pipermail/vtk-developers/2016-May/033505.html). Reading them is worthwhile IMHO.
> 
> 
> Good job.

Thank you very much @jhlegarreta for your detailed review. I will definitely keep these pointers in my mind.

**Pull request merged**